### PR TITLE
Fix race condition in EnC implementation that might cause VS crash

### PR DESCRIPTION
**Pull request by tmat**
**Customer scenario**

VS occasionally crashes while editing code during debugging.

**Bugs this fixes:**

VSO [403661](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/403661)

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

None.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

The code that unmarshalls COM PDB reader isn't thread-safe but could potentially be invoked on multiple threads in parallel when the compiler emits the IL delta and queries the PDB.

**How was the bug found?**

Watson.

### CODE REVIEW

**Comment by tmat** [comment]
@ivanbasov

File: src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs [hunk 1]
```diff
@@ -1309,7 +1293,7 @@ public int GetCurrentExceptionSpanPosition(uint exceptionRegionId, VsTextSpan[]
             }
         }
 
-        private static int MarshalPdbReader(IDebugUpdateInMemoryPE2 updater, out IntPtr pdbReaderPointer, out ISymUnmanagedReader3 managedSymReader)
+        private Lazy<ISymUnmanagedReader3> MarshalPdbReader(IDebugUpdateInMemoryPE2 updater)
```
Comment (ivanbasov) [inline src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs]: >private Lazy<ISymUnmanagedReader3>  [](start = 8, length = 35)

can still be static

**Comment by ivanbasov** [comment]
:shipit:

**Comment by tmat** [comment]
@MattGertz for ask mode approval

**Comment by tmat** [comment]
windows_release_unit64_prtest failure: https://github.com/dotnet/roslyn/issues/19918

**Comment by tmat** [comment]
windows_coreclr_test_prtest failure: https://github.com/dotnet/roslyn/issues/19929

**Comment by tmat** [comment]
test windows_coreclr_test_prtest please

**Pull request merged**